flink-runtime/src/test/java/org/apache/flink/runtime/deployment/ShuffleDeploymentDescriptorTest.java

```diff
@@ -0,0 +1,179 @@
+
+				assertTrue(remoteSdd instanceof UnknownShuffleDeploymentDescriptor);
+				typedUnknownSdd = (UnknownShuffleDeploymentDescriptor) remoteSdd;
+				assertEquals(remotePartitionId, typedUnknownSdd.getResultPartitionID());
+			}
+
+			assertTrue(unknownSdd instanceof UnknownShuffleDeploymentDescriptor);
+			typedUnknownSdd = (UnknownShuffleDeploymentDescriptor) unknownSdd;
+			assertEquals(unknownPartitionId, typedUnknownSdd.getResultPartitionID());
+		}
+	}
+
+	@Test
+	public void testUnknownChannelWithoutLazyDeploymentThrows() throws Exception {
+		ResultPartitionID unknownPartitionId = new ResultPartitionID();
+
+		// This should work if lazy deployment is allowed
+		ShuffleDeploymentDescriptor unknownSdd = getConsumedPartitionSdd(
+			unknownPartitionId, ExecutionState.CREATED, Collections.emptyMap(), true);
+
+		assertTrue(unknownSdd instanceof UnknownShuffleDeploymentDescriptor);
+		UnknownShuffleDeploymentDescriptor typedUnknownSdd = (UnknownShuffleDeploymentDescriptor) unknownSdd;
+		assertEquals(unknownPartitionId, typedUnknownSdd.getResultPartitionID());
+
+		try {
+			// Fail if lazy deployment is *not* allowed
+			getConsumedPartitionSdd(
+				unknownPartitionId, ExecutionState.CREATED, Collections.emptyMap(), false);
+			fail("Did not throw expected ExecutionGraphException");
+		} catch (ExecutionGraphException ignored) {
+		}
+	}
+
+	private static ShuffleDeploymentDescriptor getConsumedPartitionSdd(
+		ResultPartitionID id,
+		ExecutionState state,
+		Map<IntermediateResultPartitionID, ResultPartitionDeploymentDescriptor> producedPartitions)
+		throws ExecutionGraphException {
+
+		return getConsumedPartitionSdd(id, state, producedPartitions, true);
+	}
+
+	private static ShuffleDeploymentDescriptor getConsumedPartitionSdd(
+		ResultPartitionID id,
+		ExecutionState state,
+		Map<IntermediateResultPartitionID, ResultPartitionDeploymentDescriptor> producedPartitions,
+		boolean allowLazyDeployment)
+		throws ExecutionGraphException {
+
+		return TaskDeploymentDescriptorFactory.getConsumedPartitionSdd(
+			id, ResultPartitionType.PIPELINED, true, state, allowLazyDeployment, producedPartitions);
+	}
+
+	private static void addPartition(
+		ResultPartitionID id,
+		ResourceID location,
+		Map<IntermediateResultPartitionID, ResultPartitionDeploymentDescriptor> producedPartitions)
+		throws ExecutionException, InterruptedException {
+
+		ProducerShuffleDescriptor prsd =
+			new ProducerShuffleDescriptor(location, CONNECTION.getAddress(), id.getProducerId());
+		PartitionShuffleDescriptor psd = new PartitionShuffleDescriptor(
+			new IntermediateDataSetID(), id.getPartitionId(), ResultPartitionType.PIPELINED,
+			1, 1, 0);
+		ShuffleDeploymentDescriptor sdd =
+			DefaultShuffleMaster.getInstance().registerPartitionWithProducer(psd, prsd).get();
+		ResultPartitionDeploymentDescriptor rdd =
+			new ResultPartitionDeploymentDescriptor(psd, sdd, true);
+		producedPartitions.put(id.getPartitionId(), rdd);
```

Comment:
I think output arguments are not so nice. Especially since the method returns `void`. I would be ok with having something like `addPartition(producedPartitions, createResultPartitionDeploymentDescriptor(....))` but here we are mixing things in a less clean way.